<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GdbOptionsPage</class>
 <widget class="QWidget" name="GdbOptionsPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>398</width>
    <height>412</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBoxLocations">
     <property name="title">
      <string>GDB</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::ExpandingFieldsGrow</enum>
      </property>
      <property name="horizontalSpacing">
       <number>6</number>
      </property>
      <property name="verticalSpacing">
       <number>6</number>
      </property>
      <property name="margin">
       <number>9</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="labelGdbStartupScript">
        <property name="toolTip">
         <string>This is either empty or points to a file containing GDB commands that will be executed immediately after GDB starts up.</string>
        </property>
        <property name="text">
         <string>GDB startup script:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="Utils::PathChooser" name="scriptFileChooser" native="true"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelGdbWatchdogTimeout">
        <property name="toolTip">
         <string>This is the number of seconds Qt Creator will wait before
it terminates a non-responsive GDB process. The default value of 20 seconds
should be sufficient for most applications, but there are situations when
loading big libraries or listing source files takes much longer than that
on slow machines. In this case, the value should be increased.</string>
        </property>
        <property name="text">
         <string>GDB timeout:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="spinBoxGdbWatchdogTimeout">
        <property name="toolTip">
         <string>This is the number of seconds Qt Creator will wait before
it terminates a non-responsive GDB process. The default value of 20 seconds
should be sufficient for most applications, but there are situations when
loading big libraries or listing source files takes much longer than that
on slow machines. In this case, the value should be increased.</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="minimum">
         <number>20</number>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
        <property name="singleStep">
         <number>20</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxSkipKnownFrames">
        <property name="toolTip">
         <string>Allows 'Step Into' to compress several steps into one step for less noisy debugging. For example, the atomic reference
 counting code is skipped, and a single 'Step Into' for a signal emission ends up directly in the slot connected to it.</string>
        </property>
        <property name="text">
         <string>Skip known frames when stepping</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxUseMessageBoxForSignals">
        <property name="toolTip">
         <string>This will show a message box as soon as your application receives a signal like SIGSEGV during debugging.</string>
        </property>
        <property name="text">
         <string>Show a message box when receiving a signal</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxAdjustBreakpointLocations">
        <property name="toolTip">
         <string>GDB allows setting breakpoints on source lines for which no code was generated. In such situations the breakpoint is shifted to the next source code line for which code was actually generated. This option reflects such temporary change by moving the breakpoint markers in the source code editor.</string>
        </property>
        <property name="text">
         <string>Adjust breakpoint locations</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxLoadGdbInit">
        <property name="toolTip">
         <string>This allows or inhibits reading the user's default .gdbinit file on debugger startup.</string>
        </property>
        <property name="text">
         <string>Load .gdbinit file on startup</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxTargetAsync">
        <property name="text">
         <string>Use asynchronous mode to control the inferior</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QCheckBox" name="checkBoxEnableReverseDebugging">
        <property name="toolTip">
         <string>Enable reverse debugging Selecting this enables reverse debugging. NOTE: This feature is very slow and unstable on the GDB side. It exhibits unpredictable behaviour when going backwards over system calls and is very likely to destroy your debugging session.</string>
        </property>
        <property name="text">
         <string>Enable reverse debugging</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBoxPluginDebugging">
     <property name="title">
      <string>Behavior of Breakpoint Setting in Plugins</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QRadioButton" name="radioButtonAllPluginBreakpoints">
        <property name="toolTip">
         <string>This is the slowest but safest option.</string>
        </property>
        <property name="text">
         <string>Try to set breakpoints in plugins always automatically</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="radioButtonSelectedPluginBreakpoints">
        <property name="text">
         <string>Try to set breakpoints in selected plugins</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>10</width>
            <height>10</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="labelSelectedPluginBreakpoints">
          <property name="text">
           <string>Matching regular expression: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditSelectedPluginBreakpointsPattern"/>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="radioButtonNoPluginBreakpoints">
        <property name="text">
         <string>Never set breakpoints in plugins automatically</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::MinimumExpanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>10</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Utils::PathChooser</class>
   <extends>QWidget</extends>
   <header location="global">utils/pathchooser.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
